spring-cloud-gcp-data-datastore/src/main/java/org/springframework/cloud/gcp/data/datastore/core/LazyUtil.java

```diff
@@ -79,23 +99,30 @@ private static SimpleLazyDynamicInvocationHandler getProxy(Object object) {
 			return (SimpleLazyDynamicInvocationHandler) Proxy
 					.getInvocationHandler(object);
 		}
+		else if (object instanceof Factory) {
+			Callback[] callbacks = ((Factory) object).getCallbacks();
+			if (callbacks != null && callbacks.length == 1
+					&& callbacks[0] instanceof SimpleLazyDynamicInvocationHandler) {
+				return (SimpleLazyDynamicInvocationHandler) callbacks[0];
+			}
```

Comment:
Should we log a warning if callbacks length is > 1? Is this a "this should never happen" scenario?

Reply:
This should never happen if it's our proxy.